web/yo/app/scripts/controllers/genes.js

```diff
@@ -131,6 +131,85 @@ angular.module('oncokbApp')
                     });
                 }
             };
+            $scope.setUpdateTime = function () {
+                fetchGeneDoc(0);
+            };
+            function fetchGeneDoc(index) {
+                if(index < $scope.documents.length) {
+                    var fileId = $scope.documents[index].id;
+                    storage.getRealtimeDocument(fileId).then(function(realtime) {
+                        if (realtime && realtime.error) {
+                            console.log('did not get realtime document.');
+                            $timeout(function() {
+                                fetchGeneDoc(++index);
+                            },    2000, false);
+                        } else {
+                            $scope.docUpdateTime = new Date($scope.documents[index].modifiedDate).getTime();
+                            $scope.geneRealtimeModel = realtime.getModel();
+                            $scope.geneDoc = realtime.getModel().getRoot().get('gene');
+                            console.log(index, $scope.geneDoc.name.getText());
+                            setUpdateTimeGene();
+                            $timeout(function() {
+                                fetchGeneDoc(++index);
+                            },    2000, false);
+                        }
+                    });
+                } else {
+                    console.log('It is finished.');
+                }
+            }
+            function checkAndSet(reviewObj) {
+                if(!reviewObj) {
+                    reviewObj = $scope.geneRealtimeModel.createMap();
+                    console.log($scope.geneDoc.name.getText(), 'new map assigned');
+                }
+                if(!reviewObj.get('updateTime')) {
+                    reviewObj.set('updateTime', $scope.docUpdateTime);
+                    console.log('set time');
+                }
+            }
+            function setUpdateTimeGene() {
+                $scope.geneRealtimeModel.beginCompoundOperation();
+                checkAndSet($scope.geneDoc.summary_review);
+                checkAndSet($scope.geneDoc.type_review);
+                checkAndSet($scope.geneDoc.background_review);
+                for (var i = 0; i < $scope.geneDoc.mutations.length; i++) {
+                    var mutation = $scope.geneDoc.mutations.get(i);
+                    checkAndSet(mutation.name_review);
+                    checkAndSet(mutation.oncogenic_review);
+                    checkAndSet(mutation.shortSummary_review);
```

Comment:
yes, right. This script was written before we combine two sections. I will double check and update it.